### PR TITLE
add contributing guidelines

### PR DESCRIPTION
**Pull request by 27149chen**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+2. Run `make test` in the project root directory.
+
+## Developing rules
+1.  If CRDs in `deploy/crds` are updated, you need to update the file with the same name in `deploy/olm-catalog/ibm-block-csi-operator` accordingly.
+
+2.  If `role.yaml`, `role_binding.yaml` or `operator.yaml` is in `deploy` are updated, you need to update the ClusterServiceVersion(CSV) file in `deploy/olm-catalog/ibm-block-csi-operator` accordingly.
+
+3. If README.md is updated, you need to update the ClusterServiceVersion(CSV) file in `deploy/olm-catalog/ibm-block-csi-operator` accordingly.
+
+4. Make sure to run `make update` or `make test` before you finish a commit.
```
Comment (shay-berman) [inline CONTRIBUTING.md]: move this to (3) so it will be clear that make update do this work

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+- [go](https://golang.org/dl/) version v1.13+.
+- [docker](https://docs.docker.com/install/) version 17.03+.
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) version v1.14.1+.
+- [kubebuilder](https://book.kubebuilder.io/quick-start.html#installation) v2.0.1+
+
+## How to run unit tests locally
+1. Make sure kubebuilder is installed.
+2. Run `make test` in the project root directory.
+
+## Developing rules
```
Comment (shay-berman) [inline CONTRIBUTING.md]: rules -> guideline

File: CONTRIBUTING.md (hunk 1 above)
Comment (shay-berman) [inline CONTRIBUTING.md]: please add bullet about how to generate the OLM and also how to generate the operator SDK (and if and when we need to run it again for updates)

Comment (shay-berman) [reply CONTRIBUTING.md]: You can also mentioned that node_agent is currently disabled and maybe how to enable it back (if we will decide so later on).

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+1.  If CRDs in `deploy/crds` are updated, you need to update the file with the same name in `deploy/olm-catalog/ibm-block-csi-operator` accordingly.
```
Comment (rivkap) [inline CONTRIBUTING.md]: If CRDs in `deploy/crds` are updated, the same file names located in `deploy/olm-catalog/ibm-block-csi-operator` must be updated accordingly.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+2.  If `role.yaml`, `role_binding.yaml` or `operator.yaml` is in `deploy` are updated, you need to update the ClusterServiceVersion(CSV) file in `deploy/olm-catalog/ibm-block-csi-operator` accordingly.
```
Comment (rivkap) [inline CONTRIBUTING.md]: If `role.yaml`, `role_binding.yaml`, or `operator.yaml` in `deploy` are updated, the ClusterServiceVersion(CSV) file in `deploy/olm-catalog/ibm-block-csi-operator` must be updated accordingly.

File: CONTRIBUTING.md (hunk 1 above)
Comment (rivkap) [inline CONTRIBUTING.md]: 3. If README.md is updated, ClusterServiceVersion(CSV) file in `deploy/olm-catalog/ibm-block-csi-operator` must be updated accordingly.
IMPORTANT: Be sure to run `make update` or `make test` before you finish a commit.

Comment (27149chen) [reply CONTRIBUTING.md]: > You can also mentioned that node_agent is currently disabled and maybe how to enable it back (if we will decide so later on).

I think it is unnecessary to mention that.

Comment (27149chen) [reply CONTRIBUTING.md]: > please add bullet about how to generate the OLM and also how to generate the operator SDK (and if and when we need to run it again for updates)

added

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+- [docker](https://docs.docker.com/install/) version 17.03+.
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) version v1.14.1+.
+- [kubebuilder](https://book.kubebuilder.io/quick-start.html#installation) v2.0.1+
+
+## How to run unit tests locally
+1. Make sure kubebuilder is installed.
+2. Run `make test` in the project root directory.
+
+## Developing guidelines
+1. If CRDs in `deploy/crds` are updated, the same file names located in `deploy/olm-catalog/ibm-block-csi-operator` must be updated accordingly.
```
Comment (shay-berman) [inline CONTRIBUTING.md]: 1. what about olm regenerate - is there any process for that?
2. mentioned that a change in olm directory should be pushed to operatorhub (maybe add link or referance)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+2. If `role.yaml`, `role_binding.yaml`, or `operator.yaml` in `deploy` are updated, the ClusterServiceVersion(CSV) file in `deploy/olm-catalog/ibm-block-csi-operator` must be updated accordingly.
+3. If `README.md` is updated, ClusterServiceVersion(CSV) file in `deploy/olm-catalog/ibm-block-csi-operator` must be updated accordingly.
+4. Be sure to run `make update` or `make test` before you finish a commit.
```
Comment (rivkap) [inline CONTRIBUTING.md]: Can this be as a note for step #3? I don't think this should be a separate step

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+5. Run `operator-sdk add` to add a new api or controller, for more details, please refer to https://github.com/operator-framework/operator-sdk
```
Comment (rivkap) [inline CONTRIBUTING.md]: Run `operator-sdk add` to add a new API or controller, for more details, please refer to https://github.com/operator-framework/operator-sdk.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+6. Run `operator-sdk generate k8s` and `operator-sdk generate openapi` after you change something in `pkg/apis`.
```
Comment (rivkap) [inline CONTRIBUTING.md]: I don't think this should be a separate step - I think that it should be a separate note at the end of the process:
NOTE: After changes are made in 'pkg/apis', be sure to run `operator-sdk generate k8s` and `operator-sdk generate openapi`

File: CONTRIBUTING.md (hunk 6 above)
Comment (27149chen) [reply CONTRIBUTING.md]: But it is not related to #3, and all of them are tips, not a step by step thing, what do you think?

File: CONTRIBUTING.md (hunk 8 above)
Comment (27149chen) [reply CONTRIBUTING.md]: same as above, all of them are tips, what do you think?

Comment (rivkap) [reply CONTRIBUTING.md]: I understand now what you're saying.
If that is the case, I think instead of numbered steps (1, 2, 3....) make them bullets.

Comment (27149chen) [reply CONTRIBUTING.md]: done, please have a look again.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -11,12 +11,12 @@
...
-3. If `README.md` is updated, ClusterServiceVersion(CSV) file in `deploy/olm-catalog/ibm-block-csi-operator` must be updated accordingly.
-4. Be sure to run `make update` or `make test` before you finish a commit.
-5. Run `operator-sdk add` to add a new API or controller, for more details, please refer to https://github.com/operator-framework/operator-sdk.
-6. Run `operator-sdk generate k8s` and `operator-sdk generate openapi` after you change something in `pkg/apis`.
+- If CRDs in `deploy/crds` are updated, the same file names located in `deploy/olm-catalog/ibm-block-csi-operator` must be updated accordingly.
+- If `role.yaml`, `role_binding.yaml`, or `operator.yaml` in `deploy` are updated, the ClusterServiceVersion(CSV) file in `deploy/olm-catalog/ibm-block-csi-operator` must be updated accordingly.
+- If `README.md` is updated, ClusterServiceVersion(CSV) file in `deploy/olm-catalog/ibm-block-csi-operator` must be updated accordingly.
+- Be sure to run `make update` or `make test` before you finish a commit.
```
Comment (rivkap) [inline CONTRIBUTING.md]: I like this better. I would make the "Be sure to...." bullet to be first in the list.
Other than that I approve

**Pull request merged**